### ISSUE TITLE
Mypy error messages are truncated if mypy config has `pretty = true`

### ISSUE THREAD

**Issue by spookylukey** [flycheck/flycheck#1983]
## Checklist
- [X] I have checked existing issues for potential duplicates before creating this one.
- [X] I have read the [Troubleshooting guide][].

## Bug description
If mypy config has `pretty = true`, then messages appear truncated. For example, for this file:
```python
def myfunc(a: int, b: str) -> int:
    return a


def foo() -> None:
    print(myfunc(1, 1))
```

Normally flycheck shows this:
```
Argument 2 to "myfunc" has incompatible type "int"; expected "str" (python-mypy)
```

If you add `pretty = true` to mypy.ini, however, you get this:

```
Argument 2 to "myfunc" has incompatible (python-mypy)
```

In some cases it gets truncated to just two or three words of the original message, making it very hard to decipher.

## Steps to reproduce
As described.

- [X] I have read https://emacs.stackexchange.com/questions/28429/how-do-i-troubleshoot-emacs-problems

## Expected behavior
You should get the full error message regardless of the the `pretty` setting.

## System configuration
```
Syntax checkers for buffer flycheck-mypy-bug.py in python-mode:

First checker to run:

  python-flake8
    - may enable:         yes
    - executable:         Found at /home/luke/devel/clients/datapane/datapane-hosted/dp-server/.venv/bin/flake8
    - configuration file: Not found
    - next checkers:      python-pylint, python-mypy

Checkers that may run as part of the first checker's chain:

  python-mypy
    - may enable:         yes
    - may run:            t
    - executable:         Found at /home/luke/devel/clients/datapane/datapane-hosted/dp-server/.venv/bin/mypy
    - configuration file: Found at "/home/luke/devel/clients/datapane/datapane-hosted/dp-server/mypy.ini"

Checkers that could run if selected:

  python-pycompile  select
    - may enable:    yes
    - executable:    Found at /home/luke/devel/clients/datapane/datapane-hosted/dp-server/.venv/bin/python3
    - next checkers: python-mypy

  python-pyright  select
    - may enable: yes
    - executable: Found at /home/luke/.local/bin/pyright

Checkers that are compatible with this mode, but will not run until properly configured:

  python-coverage
    - may enable: yes
    - may run:    nil
    - report:     Could not find coverage file. (Hint: use ‘M-x python-coverage-select-coverage-file’ to choose manually.)

The following syntax checkers are not registered:
  - bazel-build-buildifier
  - bazel-module-buildifier
  - bazel-starlark-buildifier
  - bazel-workspace-buildifier
  - python-pylint
Try adding these syntax checkers to `flycheck-checkers'.

Flycheck Mode is enabled. Use C-u C-c ! x to enable disabled checkers.

--------------------

Flycheck version: 33snapshot (package: 20221104.549)
Emacs version:    28.1
System:           x86_64-pc-linux-gnu
Window system:    x
```

I've also verified this with a minimal setup ("emacs -q") loading just flycheck.el (and immediate dependencies), and also checked against Emacs 28.2, all with the same behaviour.

## Fix
A simple fix is to change the invocation of mypy to include ``--no-pretty``, which overrides the config file value:

```elisp
;; flycheck.el
(flycheck-define-checker python-mypy
  "Mypy syntax and type checker.  Requires mypy>=0.580.

See URL `http://mypy-lang.org/'."
  :command ("mypy"
            "--show-column-numbers"
            "--no-pretty"   ;; ADD THIS
            (config-file "--config-file" flycheck-python-mypy-config)
            (option "--cache-dir" flycheck-python-mypy-cache-dir)
            source-original)
  :error-patterns
  ((error line-start (file-name) ":" line (optional ":" column)
          ": error:" (message) line-end)
   (warning line-start (file-name) ":" line (optional ":" column)
            ": warning:" (message) line-end)
   (info line-start (file-name) ":" line (optional ":" column)
         ": note:" (message) line-end))
  :working-directory flycheck-python-find-project-root
  :modes python-mode
  ;; Ensure the file is saved, to work around
  ;; https://github.com/python/mypy/issues/4746.
  :predicate flycheck-buffer-saved-p)
```
I've tested this and it works. I can prepare a PR.
